src/cryptonote_core/miner.cpp

```diff
@@ -365,6 +398,11 @@ namespace cryptonote
         misc_utils::sleep_no_w(100);
         continue;
       }
+      else if( m_is_background_mining_enabled )
+      {
+        misc_utils::sleep_no_w(m_miner_extra_sleep);
```

Comment:
This (semi) substantial thread change is by no means "mandatory". We can get background mining in first, then I can attempt another patch at this for a separate review. I'm not sure on the process for that, so @moneromooo-monero  or @fluffypony would have to comment on that.